plugins/modules/net_tools/ldap/ldap_search.py

```diff
@@ -0,0 +1,196 @@
+- community.general.ldap_search:
+    dn: "ou=groups,dc=example,dc=com"
+  register: ldap_groups
+
+# Return GIDs for all groups
+- community.general.ldap_search:
+    dn: "ou=groups,dc=example,dc=com"
+    scope: "onelevel"
+    attrs:
+      - "gidNumber"
+  register: ldap_group_gids
+"""
+
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib, BOOLEANS
+from ansible.module_utils._text import to_native, to_bytes
+from ansible_collections.community.general.plugins.module_utils.ldap import LdapGeneric, gen_specs
+
+LDAP_IMP_ERR = None
+try:
+    import ldap
+
+    HAS_LDAP = True
+except ImportError:
+    LDAP_IMP_ERR = traceback.format_exc()
+    HAS_LDAP = False
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=gen_specs(
+            dn=dict(type='str', required=True),
+            scope=dict(type='str', default='base', choices=['base', 'onelevel', 'subordinate', 'children']),
+            filter=dict(type='str', default='(objectClass=*)'),
+            attrs=dict(type='list'),
+            schema=dict(type='bool', default=False),
+        ),
+        supports_check_mode=True,
+    )
+
+    if not HAS_LDAP:
+        module.fail_json(msg=missing_required_lib('python-ldap'),
+                         exception=LDAP_IMP_ERR)
+
+    if not module.check_mode:
+        try:
+            LdapSearch(module).main()
+        except Exception as exception:
+            module.fail_json(msg="Attribute action failed.", details=to_native(exception))
+
+    module.exit_json(changed=True)
+
+
+def _extract_entry(dn, attrs):
+    extracted = {'dn': dn}
+    for attr, val in list(attrs.items()):
+        if len(val) == 1:
+            extracted[attr] = val[0]
+        else:
+            extracted[attr] = val
+    return extracted
+
+
+class LdapSearch(LdapGeneric):
+    def __init__(self, module):
+        LdapGeneric.__init__(self, module)
+
+        self.dn = self.module.params['dn']
+        self.filterstr = self.module.params['filter']
+        self.attrlist = []
+        self._load_scope()
+        self._load_attrs()
+        self._load_schema()
+
+    def _load_schema(self):
+        self.schema = self.module.boolean(self.module.params['schema'])
+        if self.schema:
+            self.attrsonly = 1
+        else:
+            self.attrsonly = 0
+
+    def _load_scope(self):
+        scope = self.module.params['scope']
+        if scope == 'base':
+            self.scope = ldap.SCOPE_BASE
+        elif scope == 'onelevel':
+            self.scope = ldap.SCOPE_ONELEVEL
+        elif scope == 'subordinate':
+            self.scope = ldap.SCOPE_SUBORDINATE
+        elif scope == 'children':
+            self.scope = ldap.SCOPE_SUBTREE
+        else:
+            self.module.fail_json(msg="scope must be one of: base, onelevel, subordinate, children")
+
+    def _load_attrs(self):
+        if not self.module.params['attrs'] or self.module.params['attrs'] is None:
```

Comment:
You can replace the whole method body with one line:
```suggestion
        return self.module.params['attrs'] or None
```